eng/pipelines/templates/BuildAndTest.yml

```diff
@@ -22,16 +22,48 @@ steps:
   - script: ${{ parameters.buildScript }}
             -restore -build
             -configuration ${{ parameters.buildConfig }}
+            /p:ArchiveTests=true
             /bl:${{ parameters.repoLogPath }}/build.binlog
             $(_OfficialBuildIdArgs)
     displayName: Build
 
   - ${{ if ne(parameters.skipTests, 'true') }}:
     - script: ${{ parameters.dotnetScript }} dotnet-coverage collect
               --settings $(Build.SourcesDirectory)/eng/CodeCoverage.config
-              --output ${{ parameters.repoTestResultsPath }}/$(Agent.Os)_$(Agent.JobName).cobertura.xml
-              "${{ parameters.buildScript }} -test -configuration ${{ parameters.buildConfig }} /bl:${{ parameters.repoLogPath }}/tests.binlog $(_OfficialBuildIdArgs)"
-      displayName: Run tests
+              --output ${{ parameters.repoTestResultsPath }}/NonHelixTests.cobertura.xml
+              "${{ parameters.buildScript }} -testnobuild -test -configuration ${{ parameters.buildConfig }} /bl:${{ parameters.repoLogPath }}/tests.binlog $(_OfficialBuildIdArgs)"
+      displayName: Run non-helix tests
+
+    - template: /eng/common/templates/steps/send-to-helix.yml
+      parameters:
+        HelixProjectPath: '$(Build.SourcesDirectory)/tests/send-to-helix.proj'
+        HelixProjectArguments: /p:RunWithCodeCoverage=true /p:RepoTestResultsPath=${{ parameters.repoTestResultsPath }}
+
+        ${{ if eq(parameters.isWindows, 'true') }}:
+          HelixTargetQueues: Windows.11.Amd64.Client.Open
+        ${{ if ne(parameters.isWindows, 'true') }}:
+          HelixTargetQueues: Ubuntu.2204.Amd64.Open
+
+        IsWindows: ${{ parameters.isWindows }}
+        Creator: $(Build.DefinitionName)
+        HelixBuild: $(Build.BuildNumber)
+        HelixAccessToken: $(HelixApiAccessToken)
+
+    - ${{ if eq(parameters.isWindows, 'true') }}:
+      - pwsh: |
+            $_cov_filelist=$((Get-ChildItem -Path ${{ parameters.repoTestResultsPath }} -Recurse -Filter *.cobertura.xml -File | % { $_.FullName }) -join ' ')
+            echo "${{ parameters.dotnetScript }} dotnet-coverage merge $_cov_filelist -f cobertura -o ${{ parameters.repoTestResultsPath }}/$(Agent.Os)_$(Agent.JobName).cobertura.xml" > merge-cov.cmd
+            cmd /c merge-cov.cmd
+            rm merge-cov.cmd
+        displayName: Merge code coverage results
+
+    - ${{ if ne(parameters.isWindows, 'true') }}:
```

Comment:
Why do we need to split here? Why can't the same script run on both windows and non-windows?

Might need a comment explaining why we are doing this.

Reply:
I ran into my lack of pwsh skills :D

Reply:
Looking at the yml that generates the report, it handles merging already. So, this can be simplified.